src/EFCore.PG.Trgm/Infrastructure/Internal/NpgsqlTrgmOptionsExtension.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Microsoft.Extensions.DependencyInjection;
+
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Query;
+
+using Npgsql.EntityFrameworkCore.PostgreSQL.Query.ExpressionTranslators.Internal;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Infrastructure.Internal
+{
+    public class NpgsqlTrgmOptionsExtension : IDbContextOptionsExtension
+    {
+        public DbContextOptionsExtensionInfo Info => new ExtensionInfo(this);
+
+        public virtual void ApplyServices(IServiceCollection services)
+        {
+            services.AddEntityFrameworkNpgsqlTrgm();
+        }
+
+        public virtual void Validate(IDbContextOptions options)
+        {
+            var internalServiceProvider = options.FindExtension<CoreOptionsExtension>()?.InternalServiceProvider;
+            if (internalServiceProvider != null)
+            {
+                using (var scope = internalServiceProvider.CreateScope())
+                {
+                    if (scope.ServiceProvider.GetService<IEnumerable<IMethodCallTranslatorPlugin>>()
+                            ?.Any(s => s is NpgsqlTrgmMethodCallTranslatorPlugin) != true)
+                    {
+                        throw new InvalidOperationException($"{nameof(NpgsqlTrgmDbContextOptionsBuilderExtensions.UseTrgm)} requires {nameof(NpgsqlTrgmServiceCollectionExtensions.AddEntityFrameworkNpgsqlTrgm)} to be called on the internal service provider used.");
+                    }
+                }
```

Comment:
nit: In C# 8 using declarations were introduced. Let's use them.

```csharp
using var scope = internalServiceProvider.CreateScope();
var plugins = scope.ServiceProvider.GetService<IEnumerable<IMethodCallTranslatorPlugin>>();
// the following line is up to you, but Shay prefers this as I know
if (plugins is null || !plugins.Any(s => s is NpgsqlTrgmMethodCallTranslatorPlugin))
{
    throw new InvalidOperationException($"{nameof(NpgsqlTrgmDbContextOptionsBuilderExtensions.UseTrgm)} requires {nameof(NpgsqlTrgmServiceCollectionExtensions.AddEntityFrameworkNpgsqlTrgm)} to be called on the internal service provider used.");
}
```

Reply:
24ff45d1edc00f0f4748be34ee18e0c0eee25f3e fixes this.